boa/src/builtins/json/mod.rs

```diff
@@ -65,10 +67,68 @@ pub fn parse(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue
 ///
 /// [spec]: https://tc39.es/ecma262/#sec-json.stringify
 /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
-pub fn stringify(_: &mut Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
-    let obj = args.get(0).expect("cannot get argument for JSON.stringify");
-    let json = obj.to_json().to_string();
-    Ok(Value::from(json))
+pub fn stringify(_: &mut Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let object = match args.get(0) {
```

Comment:
Can we replace this with an `if let`? Should improve readability.

Reply:
@Razican I changed it, but I'd have to say that I think the match is better on readability. The else block, IMO, is too far removed from the `if`.

Reply:
Reverted back to match because there were some uncovered branches that needed to be added. Let me know your thoughts.